test/OpenTelemetry.Instrumentation.ConfluentKafka.Tests/OpenTelemetryConsumerBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using Confluent.Kafka;
+using Xunit;
+
+namespace OpenTelemetry.Instrumentation.ConfluentKafka.Tests;
+
+public class OpenTelemetryConsumerBuilderExtensionsTests
+{
+    [Fact]
+    public void ShouldConvertConsumerBuilderToInstrumentedConsumerBuilder()
+    {
+        // Arrange
+        var config = new List<KeyValuePair<string, string>>
+        {
+            new("bootstrap.servers", "localhost:9092"),
+        };
+
+        var consumerBuilder = new ConsumerBuilder<string, string>(config);
+
+        IDeserializer<string> keyDeserializer = Deserializers.Utf8;
+        IDeserializer<string> valueDeserializer = Deserializers.Utf8;
+
+        consumerBuilder.SetErrorHandler(ErrorHandler);
+        consumerBuilder.SetLogHandler(LogHandler);
+        consumerBuilder.SetStatisticsHandler(StatisticsHandler);
+        consumerBuilder.SetOAuthBearerTokenRefreshHandler(OAuthBearerTokenRefreshHandler);
+        consumerBuilder.SetOffsetsCommittedHandler(OffsetsCommittedHandler);
+        consumerBuilder.SetPartitionsAssignedHandler(PartitionsAssignedHandler);
+        consumerBuilder.SetPartitionsRevokedHandler(PartitionsRevokedHandler);
+        consumerBuilder.SetPartitionsLostHandler(PartitionsLostHandler);
+        consumerBuilder.SetKeyDeserializer(keyDeserializer);
+        consumerBuilder.SetValueDeserializer(valueDeserializer);
```

Comment:
are we saying these are mandatory ? and if yes, why not fallback ?

Reply:
That's not mandatory. Here the test just check that all the properties that have been set on the inner builder are set to the outer builder. 
Basilcally this is a decorator on a concrete type not exposing an interface.

Reply:
ok got it... Got confused where you are throwing exception, but that's if the fieldName (or expected object) is not found, it has nothing to do with value being null or not.